File: langs/python3/problems/load_balancer/stub.py
class Simulation:
    def __init__(self):
        pass

    def add_backend(self, backend_id):
        """Register a backend. "true" if created, "false" if it exists."""
        raise NotImplementedError

    def route(self):
        """Next backend id, or ""."""
        raise NotImplementedError

    def set_health(self, backend_id, flag):
        """1 up, 0 down. "true" or "invalid_request"."""
        raise NotImplementedError

    def set_weight(self, backend_id, weight):
        """Turns per cycle. "true" or "invalid_request"."""
        raise NotImplementedError

    def sticky(self, client_id):
        """Same healthy backend for this client, or ""."""
        raise NotImplementedError

    def done(self, backend_id):
        """Drop one in-flight. "true" or "invalid_request"."""
        raise NotImplementedError

File: langs/java/problems/load_balancer/stub.java
public class Simulation {
    public Simulation() {}

    /**
     * Register a backend. {@code "true"} if created, {@code "false"} if it
     * exists.
     */
    public String addBackend(String backendId) {
        return "";
    }

    /**
     * Next backend id, or {@code ""}.
     */
    public String route() {
        return "";
    }

    /**
     * 1 up, 0 down. {@code "true"} or {@code invalid_request}.
     */
    public String setHealth(String backendId, int flag) {
        return "";
    }

    /**
     * Turns per cycle. {@code "true"} or {@code invalid_request}.
     */
    public String setWeight(String backendId, int weight) {
        return "";
    }

    /**
     * Same healthy backend for this client, or {@code ""}.
     */
    public String sticky(String clientId) {
        return "";
    }

    /**
     * Drop one in-flight. {@code "true"} or {@code invalid_request}.
     */
    public String done(String backendId) {
        return "";
    }
}
